portal/src/graphql/portal/mutations/createDomainMutation.ts

```diff
@@ -0,0 +1,53 @@
+import { useCallback } from "react";
+import { gql, useMutation } from "@apollo/client";
+
+import { client } from "../../portal/apollo";
+import { CreateDomainMutation } from "./__generated__/CreateDomainMutation";
+
+const createDomainMutation = gql`
+  mutation CreateDomainMutation($appID: ID!, $domain: String!) {
+    createDomain(input: { appID: $appID, domain: $domain }) {
+      app {
+        id
+      }
+      domain {
+        id
+        createdAt
+        domain
+        apexDomain
+        isCustom
+        isVerified
+        verificationDNSRecord
+      }
+    }
+  }
+`;
+
+export function useCreateDomainMutation(
+  appID: string
+): {
+  createDomain: (domain: string) => Promise<boolean>;
+  loading: boolean;
+  error: unknown;
+} {
+  const [mutationFunction, { error, loading }] = useMutation<
+    CreateDomainMutation
+  >(createDomainMutation, {
+    client,
+    // TODO: backend return whole list of domains so apollo can
+    // automatically update the domain list
+    refetchQueries: ["DomainsQuery"],
```

Comment:
You should be able to re-select the whole list of domain in the `app` field of payload.

Reply:
It works